### PR TITLE
feat: full sync for contributor email

### PR DESCRIPTION
**Pull request by Mini256**
## CheckList
- [x] feat: full sync for contributor email
- [x] feat: increamental sync for contributor email
- [ ] feat: avoid repeated processing contributor email when pr merged
- [x] docs: add doc

### CODE REVIEW

**Comment by Mini256** [comment]
hold for fix a bug

File: src/services/ContributorService.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    // Obtain the email address from the signature of the commit author.
+    // For example:
+    // From: zhangsan <zhangsan@example.com>
+    let fromMatches = patch?.match(/From:.*/);
+
+    if (fromMatches !== null) {
+      for (let fromMatch of fromMatches) {
+        let matches = fromMatch.match(
+          /\<(\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*)\>/
```
Comment (hi-rustin) [inline src/services/ContributorService.ts:87]: Better make it be a const and explain it.

**Review by hi-rustin — APPROVED** [review]
rest LGTM.

File: src/services/ContributorService.ts (hunk 1 above)
Comment (Mini256) [reply src/services/ContributorService.ts:87]: ok

**Pull request merged**